client/src/ingredients/browser-compatibility-table/index.js

```diff
@@ -127,3 +126,81 @@ export class BrowserCompatibilityTable extends Component {
     );
   }
 }
+
+export function BrowserCompatibilityTable({ data: { uri } }) {
+  const [data, setData] = useState(null);
+  const [fetchError, setFetchError] = useState(null);
+  useEffect(() => {
+    let dismounted = false;
+    let url = document.location.pathname;
+
+    // Hopefully this hack won't be necessary once we sort out our
+    // CloudFront redirect stuff.
+    url = url.replace(/index\.html$/, "");
+
+    if (!url.endsWith("/")) url += "/";
+    url += uri;
+    fetch(url)
+      .then(r => {
+        if (!r.ok) {
+          return setFetchError(r);
+        }
+        return r.json();
+      })
+      .then(data => {
+        if (!dismounted) {
+          setData(data);
+        }
+      })
+      .catch(err => {
+        setFetchError(err);
+      });
+    return () => {
+      dismounted = true;
+    };
+  }, [uri]);
+  return (
+    <>
+      <h2 id="Browser_compatibility">Browser Compatibility</h2>
+      {/* Animate this? */}
+      {!data && fetchError ? (
+        <FetchError error={fetchError} />
+      ) : (
+        <p>Loading browser compatibility table...</p>
+      )}
+      {data && <BrowserCompatibilityTableContent data={data} />}
```

Comment:
Looks like ```Loading browser compatibility table...``` will always display even after being loaded. Could probably use a little helper function to render the `BrowserCompatibilityTableContent` if data is present, otherwise render either the error message or ```Loading browser compatibility table...```

Reply:
Thanks!